File: source/kCura.Relativity.Export/Exporters/ObjectExportInfo.vb
Namespace kCura.Relativity.Export.Exports
	Public Class ObjectExportInfo
		Private _images As System.Collections.ArrayList
		Private _native As Object
		Private _totalFileSize As Int64
		Private _totalNumberOfFiles As Int64
		Private _artifactID As Int32
		Private _hasFullText As Boolean
		Private _identifierValue As String = ""
		Private _nativeExtension As String = ""
		Private _nativeFileGuid As String = ""
		Private _nativeTempLocation As String = ""
		Private _productionBeginBates As String = ""
		Private _originalFileName As String = ""
		Private _nativeSourceLocation As String = ""
		Private _hasCountedNative As Boolean = False
		Private _hasCountedTextFile As Boolean = False
		Private _docCount As Int32 = 1
		Private _fileID As Int32 = 0
		Private _metadata As Object()

		Public Property Metadata() As Object()
			Get
				Return _metadata
			End Get
			Set(ByVal value As Object())
				_metadata = value
			End Set
		End Property

		Public Property Images() As System.Collections.ArrayList
			Get
				Return _images
			End Get
			Set(ByVal value As System.Collections.ArrayList)
				_images = value
			End Set
		End Property

		Public Property Native() As Object
			Get
				Return _native
			End Get
			Set(ByVal value As Object)
				_native = value
			End Set
		End Property

		Public Property HasFullText() As Boolean
			Get
				Return _hasFullText
			End Get
			Set(ByVal value As Boolean)
				_hasFullText = value
			End Set
		End Property

		Public Property TotalFileSize() As Int64
			Get
				Return _totalFileSize
			End Get
			Set(ByVal value As Int64)
				_totalFileSize = value
			End Set
		End Property

		Public Property TotalNumberOfFiles() As Int64
			Get
				Return _totalNumberOfFiles
			End Get
			Set(ByVal value As Int64)
				_totalNumberOfFiles = value
			End Set
		End Property

		Public Property IdentifierValue() As String
			Get
				Return _identifierValue
			End Get
			Set(ByVal value As String)
				_identifierValue = value
			End Set
		End Property

		Public Property ArtifactID() As Int32
			Get
				Return _artifactID
			End Get
			Set(ByVal value As Int32)
				_artifactID = value
			End Set
		End Property

		Public Property NativeExtension() As String
			Get
				Return _nativeExtension
			End Get
			Set(ByVal value As String)
				_nativeExtension = value
			End Set
		End Property

		Public Property NativeFileGuid() As String
			Get
				Return _nativeFileGuid
			End Get
			Set(ByVal value As String)
				_nativeFileGuid = value
			End Set
		End Property

		Public Property NativeTempLocation() As String
			Get
				Return _nativeTempLocation
			End Get
			Set(ByVal value As String)
				_nativeTempLocation = value
			End Set
		End Property

		Public Property NativeSourceLocation() As String
			Get
				Return _nativeSourceLocation
			End Get
			Set(ByVal value As String)
				_nativeSourceLocation = value
			End Set
		End Property

		Public Function NativeFileName(ByVal appendToOriginal As Boolean) As String
			Dim retval As String
			If appendToOriginal Then
				retval = IdentifierValue & "_" & OriginalFileName
			Else
				If Not NativeExtension = "" Then
					retval = IdentifierValue & "." & NativeExtension
				Else
					retval = IdentifierValue
				End If
			End If
			Return kCura.Utility.File.Instance.ConvertIllegalCharactersInFilename(retval)
		End Function

		Public Function FullTextFileName(ByVal nameFilesAfterIdentifier As Boolean) As String
			Dim retval As String
			If Not nameFilesAfterIdentifier Then
				retval = Me.ProductionBeginBates
			Else
				retval = Me.IdentifierValue
			End If
			Return kCura.Utility.File.Instance.ConvertIllegalCharactersInFilename(retval & ".txt")
		End Function

		Public Property OriginalFileName() As String
			Get
				Return _originalFileName
			End Get
			Set(ByVal value As String)
				_originalFileName = value
			End Set
		End Property

		Public ReadOnly Property ProductionBeginBatesFileName(ByVal appendToOriginal As Boolean) As String
			Get
				Dim retval As String
				If appendToOriginal Then
					retval = ProductionBeginBates & "_" & OriginalFileName
				Else
					If Not NativeExtension = "" Then
						retval = ProductionBeginBates & "." & NativeExtension
					Else
						retval = ProductionBeginBates
					End If
				End If
				Return kCura.Utility.File.Instance.ConvertIllegalCharactersInFilename(retval)
			End Get
		End Property

		Public ReadOnly Property NativeCount() As Int64
			Get
				If Me.NativeFileGuid = "" Then
					If Not Me.FileID = Nothing Or Me.FileID <> 0 Then
						Return 1
					Else
						Return 0
					End If
				Else
					Return 1
				End If
			End Get
		End Property

		Public ReadOnly Property ImageCount() As Int64
			Get
				If Me.Images Is Nothing Then Return 0
				Return Me.Images.Count
			End Get
		End Property

		Public Property ProductionBeginBates() As String
			Get
				Return _productionBeginBates
			End Get
			Set(ByVal value As String)
				_productionBeginBates = value
			End Set
		End Property

		Public Property HasCountedNative() As Boolean
			Get
				Return _hasCountedNative
			End Get
			Set(ByVal value As Boolean)
				_hasCountedNative = value
			End Set
		End Property
		Public Property HasCountedTextFile() As Boolean
			Get
				Return _hasCountedTextFile
			End Get
			Set(ByVal value As Boolean)
				_hasCountedTextFile = value
			End Set
		End Property

		Public Property FileID() As Int32
			Get
				Return _fileID
			End Get
			Set(ByVal value As Int32)
				_fileID = value
			End Set
		End Property


		Public ReadOnly Property DocCount() As Int32
			Get
				Dim retval As Int32 = _docCount
				If retval = 1 Then _docCount -= 1
				Return retval
			End Get
		End Property
	End Class
End Namespace

File: source/kCura.Relativity.Export.ConvertedToC#/Exporters/ObjectExportInfo.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Linq;
namespace kCura.Relativity.Export.Exports
{
	public class ObjectExportInfo
	{
		private System.Collections.ArrayList _images;
		private object _native;
		private Int64 _totalFileSize;
		private Int64 _totalNumberOfFiles;
		private Int32 _artifactID;
		private bool _hasFullText;
		private string _identifierValue = "";
		private string _nativeExtension = "";
		private string _nativeFileGuid = "";
		private string _nativeTempLocation = "";
		private string _productionBeginBates = "";
		private string _originalFileName = "";
		private string _nativeSourceLocation = "";
		private bool _hasCountedNative = false;
		private bool _hasCountedTextFile = false;
		private Int32 _docCount = 1;
		private Int32 _fileID = 0;

		private object[] _metadata;
		public object[] Metadata {
			get { return _metadata; }
			set { _metadata = value; }
		}

		public System.Collections.ArrayList Images {
			get { return _images; }
			set { _images = value; }
		}

		public object Native {
			get { return _native; }
			set { _native = value; }
		}

		public bool HasFullText {
			get { return _hasFullText; }
			set { _hasFullText = value; }
		}

		public Int64 TotalFileSize {
			get { return _totalFileSize; }
			set { _totalFileSize = value; }
		}

		public Int64 TotalNumberOfFiles {
			get { return _totalNumberOfFiles; }
			set { _totalNumberOfFiles = value; }
		}

		public string IdentifierValue {
			get { return _identifierValue; }
			set { _identifierValue = value; }
		}

		public Int32 ArtifactID {
			get { return _artifactID; }
			set { _artifactID = value; }
		}

		public string NativeExtension {
			get { return _nativeExtension; }
			set { _nativeExtension = value; }
		}

		public string NativeFileGuid {
			get { return _nativeFileGuid; }
			set { _nativeFileGuid = value; }
		}

		public string NativeTempLocation {
			get { return _nativeTempLocation; }
			set { _nativeTempLocation = value; }
		}

		public string NativeSourceLocation {
			get { return _nativeSourceLocation; }
			set { _nativeSourceLocation = value; }
		}

		public string NativeFileName(bool appendToOriginal)
		{
			string retval = null;
			if (appendToOriginal) {
				retval = IdentifierValue + "_" + OriginalFileName;
			} else {
				if (!string.IsNullOrEmpty(NativeExtension)) {
					retval = IdentifierValue + "." + NativeExtension;
				} else {
					retval = IdentifierValue;
				}
			}
			return kCura.Utility.File.Instance.ConvertIllegalCharactersInFilename(retval);
		}

		public string FullTextFileName(bool nameFilesAfterIdentifier)
		{
			string retval = null;
			if (!nameFilesAfterIdentifier) {
				retval = this.ProductionBeginBates;
			} else {
				retval = this.IdentifierValue;
			}
			return kCura.Utility.File.Instance.ConvertIllegalCharactersInFilename(retval + ".txt");
		}

		public string OriginalFileName {
			get { return _originalFileName; }
			set { _originalFileName = value; }
		}

		public string ProductionBeginBatesFileName(bool appendToOriginal) {
		
				string retval = null;
				if (appendToOriginal) {
					retval = ProductionBeginBates + "_" + OriginalFileName;
				} else {
					if (!string.IsNullOrEmpty(NativeExtension)) {
						retval = ProductionBeginBates + "." + NativeExtension;
					} else {
						retval = ProductionBeginBates;
					}
				}
				return kCura.Utility.File.Instance.ConvertIllegalCharactersInFilename(retval);
			
		}

		public Int64 NativeCount {
			get {
				if (string.IsNullOrEmpty(this.NativeFileGuid)) {
					if (!(this.FileID == null) | this.FileID != 0) {
						return 1;
					} else {
						return 0;
					}
				} else {
					return 1;
				}
			}
		}

		public Int64 ImageCount {
			get {
				if (this.Images == null)
					return 0;
				return this.Images.Count;
			}
		}

		public string ProductionBeginBates {
			get { return _productionBeginBates; }
			set { _productionBeginBates = value; }
		}

		public bool HasCountedNative {
			get { return _hasCountedNative; }
			set { _hasCountedNative = value; }
		}
		public bool HasCountedTextFile {
			get { return _hasCountedTextFile; }
			set { _hasCountedTextFile = value; }
		}

		public Int32 FileID {
			get { return _fileID; }
			set { _fileID = value; }
		}


		public Int32 DocCount {
			get {
				Int32 retval = _docCount;
				if (retval == 1)
					_docCount -= 1;
				return retval;
			}
		}
	}
}
